### DISCUSSION TITLE
[FR] Option to transfer existing relationships to a sibling when deleting an element (Entry/Category/Tag…)

### DISCUSSION THREAD

**Discussion by rungta** [craftcms/cms#15410] [Ideas] [answered]
Craft provides a very useful option to retain author relationships when deleting a User by transferring it another User. The same option, if extended to Entries and Categories would allow easy resolution of use cases such as:
- Merging exact duplicate taxonomies (eg: two “Hiking” entries)
- Resolving duplication as a result of typos (eg: “Hiking” and “Hikking”)
- Merging similar categories (eg: “Hiking” and “Trekking”)
- etc.

![image](https://github.com/user-attachments/assets/87dc7b71-45cc-4f0d-9b76-a5339a6141f9)

**Comment by rishifter** [comment]
Hi Craft team! We've a client with a fairly large content set (around 70-80k entries at the start, and expected to grow even more over time) + a few thousand taxonomies. As a large team of authors are working on uploading the content to the site, they have quite a few duplicates starting to show up. The ability for editors/reviewers to go in and merge taxonomies while keeping the existing association with entries would be immensely beneficial for such a large content set. 

We hope this is being considered/discussed as a feature. Just dropping a message to add a bit more weight to this existing request.

**Comment by thupsi** [comment]
This could also come in handy when using the new `copy` functionality, since there's no `cut`. At least then one could copy the original entry, paste it to its new location, then delete the original and transfer the relationships to the copy.

**Comment by proimage** [comment]
I'll tack on a related enhancement:
- When creating a new Category, warn if there is a similarly-named Category that already exists.
- It would be great if the duplicate check was fuzzy enough to trigger if "Nation" exists and I'm trying to create "Country" or something. I hate to say it, but some sort of LLM would be great at this sort of matching...
- If one or more Categories are merged into another Category, it might be useful to record the Category Titles of the merged ones in the data structure of the destination Category, for use when checking for duplicates.
    - For example, if there's an existing category called "UK", I wouldn't necessarily expect creating a category called "Great Britain" to trigger the duplicate check. However, when an editor reviews things and decides to merge, say, UK -> Great Britain, it would be good if the duplicate check could "learn" from that merger (by the merged titles being recorded) and trigger if someone later tried to create "UK" or "U.K." again.
- Also, I think it would be important to add a setting—perhaps in the Category Group settings—to allow this duplicate check to be done across sites on a multisite install. I'm currently dealing with a situation where a bunch of categories were created for one site in a multisite install, even though they already existed in the system on other sites and just needed to be enabled for that one site (the categories already existed when the site was created).

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.10 is out with an improved element deletion workflow, which now shows a modal when elements are being deleted that are related by other elements, with the option to reassign those relations to another element. (https://github.com/craftcms/cms/pull/18728)
